lib/routes/podcast/podcast_clip.dart

```diff
@@ -0,0 +1,559 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:anytime/bloc/podcast/audio_bloc.dart';
+import 'package:anytime/entities/episode.dart';
+import 'package:anytime/services/audio/audio_player_service.dart';
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:provider/provider.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import 'package:breez/theme_data.dart' as breezTheme;
+import 'package:screenshot/screenshot.dart';
+import 'package:share_plus/share_plus.dart';
+
+import '../../bloc/blocs_provider.dart';
+import '../../bloc/podcast_clip/podcast_clip_bloc.dart';
+import '../../bloc/podcast_clip/podcast_clip_details_model.dart';
+import '../../theme_data.dart';
+import '../../utils/min_font_size.dart';
+import '../../widgets/network_image_builder.dart';
+
+class PodcastClipWidget extends StatelessWidget {
+  PodcastClipWidget({Key key}) : super(key: key);
+  final ScreenshotController _screenshotController = ScreenshotController();
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          GestureDetector(
+            onTap: () async {
+              final audioBloc = Provider.of<AudioBloc>(context, listen: false);
+              final podcastClipBloc =
+                  AppBlocsProvider.of<PodcastClipBloc>(context);
+              PositionState position = await getPositionDetails(audioBloc);
+              if (position != null) {
+                podcastClipBloc.setPodcastClipDetails(position: position);
+              }
+              bool canBeclipped = await podcastClipBloc.isEpisodeClipable();
+              if (!canBeclipped) {
+                return;
+              }
+              _pause(audioBloc);
+
+              showModalBottomSheet<void>(
```

Comment:
It is hard to review and maintain such a large widgets tree. Better to break this into separate widgets.

Reply:
Refractored the widget code.